### PR TITLE
feat: implement `lgt`, `llt`, `lle`,`lge`

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Continuing: https://github.com/lfortran/lfortran/pull/3570

### CODE REVIEW

**Comment by HarshitaKalani** [comment]
```fortran
program main
    character(5), allocatable :: a, b
    print*, lgt(maybe(a), b)

    contains
    function maybe(string) result(maybe_string)
        character(5), allocatable:: string
        character(len=len(string)) :: maybe_string
        maybe_string = 'works'
    end function maybe

end
```
This does not work yet.

File: src/libasr/pass/intrinsic_functions.h [hunk 1]
```diff
@@ -1384,6 +1388,142 @@ namespace Ble {
...
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_lle_" + type_to_str_python(arg_types[0]));
+        fill_func_arg("x", ASRUtils::TYPE(ASR::make_Character_t(al, loc, 1, -1, nullptr)));
+        fill_func_arg("y", ASRUtils::TYPE(ASR::make_Character_t(al, loc, 1, -1, nullptr)));
+        auto result = declare(fn_name, return_type, ReturnVar);
+        body.push_back(al, b.If(sLtE(args[0], args[1]), {
+            b.Assignment(result, b.bool32(1))
+        }, {
+            b.Assignment(result, b.bool32(0))
+        }));
```
Comment (Pranavchiku) [inline src/libasr/pass/intrinsic_functions.h:1517]: Can we not do something like?

```suggestion
	body.push_back(al, b.Assignment(result, sLtE(args[0], args[1])));
```

Comment (Pranavchiku) [reply src/libasr/pass/intrinsic_functions.h:1517]: Try this locally, if it works use it at everyplace. I am converting it to draft.

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
A small suggestion requested, also this PR needs to be rebased and conflicts resolved.

File: integration_tests/intrinsics_188.f90 [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    if (llt(lfortran, fortran) .neqv. .false.) error stop
+
+    print*, lle(hello, world)
+    if (lle(hello, world) .neqv. .true.) error stop
+    print*, lle(hello, hello)
+    if (lle(hello, hello) .neqv. .true.) error stop
+    print*, lle(lfortran, fortran)
+    if (lle(lfortran, fortran) .neqv. .false.) error stop
+    
+end
```
Comment (Shaikh-Ubaid) [inline integration_tests/intrinsics_188.f90:64]:
```suggestion
end program
```

File: integration_tests/intrinsics_188.f90 [hunk 3]
```diff
@@ -0,0 +1,64 @@
+program intrinsics_188
+        
+    character(5) :: hello = "hello"
+    character(5) :: world = "world"
+    character(8) :: lfortran = "lfortran"
+    character(7) :: fortran = "fortran"
```
Comment (Shaikh-Ubaid) [inline integration_tests/intrinsics_188.f90:6]: Let's also add couple tests when there are symbols in strings. Try including different characters with `ascii` values ranging from `0` to `127`.

**Comment by Pranavchiku** [comment]
The failure seems unrelated and can be fixed by rerun

```console
100% tests passed, 0 tests failed out of 3

Total Test time (real) =   9.93 sec
+ cd ..
+ git clean -dfx
Removing lf-fast/
Removing lf/
Removing model.dat
+ git checkout -t origin/namelist
fatal: 'origin/namelist' is not a commit and a branch 'namelist' cannot be created from it
```

**Comment by certik** [comment]
The failure is due to https://github.com/lfortran/lfortran/issues/3663.

**Review by certik — APPROVED** [review]

**Pull request merged**